x-pack/test/api_integration/apis/ml/saved_objects/initialize_trained_models.ts

```diff
@@ -0,0 +1,137 @@
+  const modelIdSpace2 = 'dfa2_classification_model_n_0';
+  const modelIdStarSpace = 'dfa3_classification_model_n_0';
+
+  async function runRequest(user: USER, expectedStatusCode: number) {
+    const { body, status } = await supertest
+      .get(`/api/ml/saved_objects/initialize`)
+      .auth(user, ml.securityCommon.getPasswordForUser(user))
+      .set(COMMON_REQUEST_HEADERS);
+    ml.api.assertResponseStatusCode(expectedStatusCode, status, body);
+
+    return body;
+  }
+
+  function getTestModel(id: string, modelType: ModelType, jobId?: string) {
+    return {
+      model_id: id,
+      body: {
+        compressed_definition: ml.api.getCompressedModelDefinition(modelType),
+        inference_config: {
+          [modelType]: {},
+        },
+        ...(jobId ? { metadata: { analytics_config: { id: jobId } } } : {}),
+        input: {
+          field_names: ['common_field'],
+        },
+      } as PutTrainedModelConfig,
+    };
+  }
+
+  describe('GET saved_objects/initialize for trained models', () => {
+    before(async () => {
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/ml/ihp_outlier');
+
+      await spacesService.create({ id: idSpace1, name: 'space_one', disabledFeatures: [] });
+      await spacesService.create({ id: idSpace2, name: 'space_two', disabledFeatures: [] });
+
+      await ml.api.initSavedObjects();
+
+      await ml.api.createDataFrameAnalyticsJob(
+        ml.commonConfig.getDFAIhpOutlierDetectionJobConfig(dfaJobId1),
+        idSpace1
+      );
+      await ml.api.updateJobSpaces(dfaJobId1, 'data-frame-analytics', [idSpace2], [], idSpace1);
+      await ml.api.assertJobSpaces(dfaJobId1, 'data-frame-analytics', [idSpace1, idSpace2]);
+
+      await ml.api.createDataFrameAnalyticsJob(
+        ml.commonConfig.getDFAIhpOutlierDetectionJobConfig(dfaJobIdStarSpace),
+        idSpace1
+      );
+
+      await ml.api.updateJobSpaces(
+        dfaJobIdStarSpace,
+        'data-frame-analytics',
+        ['*'],
+        [idSpace1],
+        idSpace1
+      );
+      await ml.api.assertJobSpaces(dfaJobIdStarSpace, 'data-frame-analytics', ['*']);
+
+      await ml.testResources.setKibanaTimeZoneToUTC();
+    });
+
+    after(async () => {
+      await spacesService.delete(idSpace1);
+      await spacesService.delete(idSpace2);
+      await ml.api.cleanMlIndices();
+      await ml.testResources.cleanMLSavedObjects();
+    });
+
+    it('should initialize trained models and inherit job spaces', async () => {
+      // create trained model linked to job, it should inherit the jobs spaces
+      const model = getTestModel(modelIdSpace1, 'classification', dfaJobId1);
+      await ml.api.createTrainedModelES(model.model_id, model.body);
+
+      const body = await runRequest(USER.ML_POWERUSER_ALL_SPACES, 200);
+
+      expect(body).to.have.property('trainedModels');
+      expect(sortBy(body.trainedModels, 'id')).to.eql([{ id: modelIdSpace1 }]);
```

Comment:
question: why do you need a `sortBy` here? It seems like there is only one item in the array

Reply:
https://github.com/elastic/kibana/pull/130115/commits/c3ae34ad2f84f91b78a6e28259e121a2e4d9d474